common/adder/adder.xml

```diff
@@ -0,0 +1,103 @@
+  <declaration>
+    <meta name="name" value="Adder" />
+    <meta name="minsize" value="50" />
+    <meta name="author" value="Corsaka" />
+    <meta name="guid" value="8f764035-18c6-4421-ac7f-5cbd327d4e78" />
+
+    <property name="Type" type="enum" default="Full" serialize="t" display="Type">
+      <option>Full</option>
+      <option>Half</option>
+    </property>
+
+    <configurations>
+      <configuration name="Full Adder" value="Type:Full" implements="full" />
+      <configuration name="Half Adder" value="Type:Half" implements="half" />
+    </configurations>
+  </declaration>
+  <connections>
+    <group conditions="horizontal">
+      <connection name="A" start="_Start-7x-30y" end="_Start-7x-60y" edge="End" />
+      <connection name="B" start="_End+7x-30y" end="_End+7x-60y" edge="End" />
+      <connection name="Sum" start="_Middle+30y" end="_Middle+60y" edge="End" />
+      <connection name="Cin" start="_Start-15x" end="_Start-45x" edge="End" />
+      <connection name="Cout" start="_End+15x" end="_End+45x" edge="End" />
+    </group>
+    <group conditions="!horizontal">
+      <connection name="A" start="_Start-35y" end="_Start-30x-35y" edge="End" />
+      <connection name="B" start="_Start+35y" end="_Start-30x+35y" edge="End" />
+      <connection name="Sum" start="_End" end="_End+30x" edge="End" />
+      <connection name="Cin" start="_Middle-45y" end="_Middle-75y" edge="End" />
+      <connection name="Cout" start="_Middle+45y" end="_Middle+75y" edge="End" />
+    </group>
+  </connections>
+  <render>
+    <group conditions="horizontal">
+      <!-- draw shape -->
+      <line start="_Start+30y" end="_Start-30x-30y" />
+      <line start="_Start+30y" end="_End+30y" />
+      <line start="_Start-30x-30y" end="_Start+15x-30y" />
+      <line start="_Start+15x-30y" end="_Middle" />
+      <line start="_Middle" end="_End-15x-30y" />
+      <line start="_End-15x-30y" end="_End+30x-30y" />
+      <line start="_End+30y" end="_End+30x-30y" />
+      <!-- draw connection lines -->
+      <line start="_Start-7x-30y" end="_Start-7x-60y" />
+      <line start="_End+7x-30y" end="_End+7x-60y" />
+      <line start="_Middle+30y" end="_Middle+60y" />
+      <line start="_Start-15x" end="_Start-45x" />
+      <line start="_End+15x" end="_End+45x" />
+      <!-- draw text -->
+      <text x="_Start-7" y="_Start-20" align="CentreCentre" size="large" value="A" />
+      <text x="_End+7" y="_End-20" align="CentreCentre" size="large" value="B" />
+      <text x="_Middle+1" y="_Middle+20" align="CentreCentre" size="large" value="SUM" />
+      <text x="_Start" y="_Middle" align="CentreCentre" size="large">
+        <value>
+          <span>C</span>
+          <sub>in</sub>
+        </value>
+      </text>
+      <text x="_End-5" y="_Middle" align="CentreCentre" size="large" value="C">
+        <value>
+          <span>C</span>
+          <sub>out</sub>
+        </value>
+      </text>
+    </group>
+    <group conditions="!horizontal">
```

Comment:
It looks like this when placed vertically:

![image](https://user-images.githubusercontent.com/5303088/75490952-01a49900-59ad-11ea-8b9c-61b2a45045b8.png)

You can see this in [GitPod](https://gitpod.io/#https://github.com/circuitdiagram/components) or VS Code:

![image](https://user-images.githubusercontent.com/5303088/75492032-cc994600-59ae-11ea-90c1-f1219da0aaa2.png)

You can add the `horizontalonly` [flag](https://componenteditor.com/help/flags/) if you only want to allow it to be placed horizontally.

Reply:
Hm. Yes. Altered to HorizontalOnly.